inception-search-mtas/src/test/java/de/tudarmstadt/ukp/inception/search/index/mtas/MtasDocumentIndexTest.java

```diff
@@ -0,0 +1,347 @@
+import de.tudarmstadt.ukp.dkpro.core.io.text.TextReader;
+import de.tudarmstadt.ukp.dkpro.core.io.text.TextWriter;
+import de.tudarmstadt.ukp.inception.search.SearchResult;
+import de.tudarmstadt.ukp.inception.search.SearchService;
+import de.tudarmstadt.ukp.inception.search.SearchServiceImpl;
+import de.tudarmstadt.ukp.inception.search.index.IndexFactory;
+import de.tudarmstadt.ukp.inception.search.index.IndexRegistry;
+import de.tudarmstadt.ukp.inception.search.index.IndexRegistryImpl;
+
+/**
+ * The Class MtasDocumentIndexTest.
+ */
+
+@RunWith(SpringRunner.class)
+@EnableAutoConfiguration
+@SpringBootTest(webEnvironment = WebEnvironment.MOCK)
+@EnableWebSecurity
+@EntityScan({ "de.tudarmstadt.ukp.clarin.webanno.model",
+        "de.tudarmstadt.ukp.clarin.webanno.security.model" })
+@TestPropertySource(locations = "classpath:MtasDocumentIndexTest.properties")
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+
+public class MtasDocumentIndexTest
+{
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    private @Autowired UserDao userRepository;
+
+    private @Autowired ProjectService projectService;
+    private @Autowired DocumentService documentService;
+    private @Autowired SearchService searchService;
+    private @Autowired AnnotationSchemaService annotationSchemaService;
+
+    // If this is not static, for some reason the value is re-set to false before a
+    // test method is invoked. However, the DB is not reset - and it should not be.
+    // So we need to make this static to ensure that we really only create the user
+    // in the DB and clean the test repository once!
+    private static boolean initialized = false;
+
+    @Before
+    public void setUp()
+    {
+        if (!initialized) {
+            userRepository.create(new User("admin", Role.ROLE_ADMIN));
+            initialized = true;
+
+            FileSystemUtils.deleteRecursively(new File("target/MtasDocumentIndexTest"));
+        }
+    }
+
+    @Test
+    public void testSimpleQuery() throws Exception
+    {
+        Project project = new Project();
+
+        project.setName("TestSimpleQuery");
+
+        projectService.createProject(project);
+
+        annotationSchemaService.initializeProject(project);
+
+        SourceDocument sourceDocument = new SourceDocument();
+
+        sourceDocument.setName("test");
+        sourceDocument.setProject(project);
+        sourceDocument.setFormat("text");
+
+        String fileContent = "The capital of Galicia is Santiago de Compostela.";
+
+        InputStream fileStream = new ByteArrayInputStream(
+                fileContent.getBytes(StandardCharsets.UTF_8));
+
+        documentService.uploadSourceDocument(fileStream, sourceDocument);
+
+        User user = userRepository.get("admin");
+        String query = "Galicia";
+
+        ArrayList<SearchResult> results = (ArrayList<SearchResult>) searchService.query(user,
```

Comment:
`List`

Reply:
ok

Reply:
Use List instead of ArrayList and remove the type cast.